src/Domain/EmbeddingMaps/Rotation.hpp

```diff
@@ -0,0 +1,177 @@
+
+/// Spatial rotation in two dimensions.
+/// Let \f$(R,\Phi)\f$ be the polar coordinates associated with
+/// \f$(\xi,\eta)\f$.
+/// Let \f$(r,\phi)\f$ be the polar coordinates associated with \f$(x,y)\f$.
+/// Applies the spatial rotation \f$\phi = \Phi + \alpha\f$.
+///
+/// The formula for the mapping is...
+/*!\f{eqnarray*}
+    x &=& \xi \cos \alpha - \eta \sin \alpha
+\\  y &=& \xi \sin \alpha + \eta \cos \alpha
+\f}.
+*/
+template <>
+class Rotation<2> : public EmbeddingMap<2, 2> {
+ public:
+  /// Constructor.
+  ///
+  /// \param rotation_angle the angle \f$\alpha\f$ (in radians).
+  explicit Rotation(double rotation_angle);
+  Rotation() = default;
+  ~Rotation() override = default;
+  Rotation(const Rotation&) = delete;
+  Rotation& operator=(const Rotation&) = delete;
+  Rotation(Rotation&&) noexcept = default;  // NOLINT
+  Rotation& operator=(Rotation&&) = delete;
+
+  std::unique_ptr<EmbeddingMap<2, 2>> get_clone() const override {
+    return std::make_unique<Rotation<2>>(rotation_angle_);
+  }
+
+  Point<2, Frame::Grid> operator()(
+      const Point<2, Frame::Logical>& xi) const override;
+
+  Point<2, Frame::Logical> inverse(
+      const Point<2, Frame::Grid>& x) const override;
+
+  double jacobian(const Point<2, Frame::Logical>& /*xi*/, size_t ud,
+                  size_t ld) const override {
+    ASSERT(0 == ld || 1 == ld, "ld = " << ld);
+    ASSERT(0 == ud || 1 == ud, "ud = " << ud);
+    return rotation_matrix_.get(ud, ld);
+  }
+
+  double inv_jacobian(const Point<2, Frame::Logical>& /*xi*/, size_t ud,
+                      size_t ld) const override {
+    ASSERT(0 == ld || 1 == ld, "ld = " << ld);
+    ASSERT(0 == ud || 1 == ud, "ud = " << ud);
+    // Recall inverse(rotation_matrix_) = transpose(rotation_matrix_)
+    return rotation_matrix_.get(ld, ud);
+  }
+
+  WRAPPED_PUPable_decl_base_template(SINGLE_ARG(EmbeddingMap<2, 2>),  // NOLINT
+                                     Rotation);                       // NOLINT
+
+  explicit Rotation(CkMigrateMessage* /* m */)
+      : rotation_angle_(std::numeric_limits<double>::signaling_NaN()),
+        rotation_matrix_() {}
+  void pup(PUP::er& p) override {  // NOLINT
+    p | rotation_angle_;
+    p | rotation_matrix_;
+  }
+
+ private:
+  double rotation_angle_{std::numeric_limits<double>::signaling_NaN()};
+  tnsr::ij<double, 2, Frame::Grid> rotation_matrix_{
+      std::numeric_limits<double>::signaling_NaN()};
+};
+
+/// Spatial rotation in three dimensions using Euler angles
+///
+/// Rotation angles should be specified in degrees.
+/// First rotation \f$\alpha\f$ is about z axis.
+/// Second rotation \f$\beta\f$ is about rotated y axis.
+/// Third rotation \f$\gamma\f$ is about rotated z axis.
+/// These rotations are of the \f$(\xi,\eta,\zeta)\f$ coordinate system with
+/// respect
+/// to the grid coordinates \f$(x,y,z)\f$.
+///
+/// The formula for the mapping is:
+/*!\f{eqnarray*}
+    x &=& \xi (\cos\gamma \cos\beta \cos\alpha - \sin\gamma \sin\alpha)
```

Comment:
I don't think we need teh `&` before _and_ after `=`, just after. Then should see how this renders, not sure if we'll need to manually break lines to have nice formatting